ci/erlang.yml

```diff
@@ -6,10 +6,14 @@ on:
   pull_request:
     branches: [ $default-branch ]
 
+permissions: read-all
+
 jobs:
 
   build:
 
+    permissions:
+      contents: read # for actions/checkout to fetch code
```

Comment:
I believe `contents: read` is a default [permission](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) even for restricted access.
So in a strict sense this change is not needed, and that keeps the workflow simple.

Reply:
My understanding is that when a new GitHub org or repo is created, the default GitHub Action token permissions are permissive. By that I mean that all permissions are granted to the `GITHUB_TOKEN`. So, if `contents: read` is not added in the workflow file, the token will get all permissions. 

Moreover, while writing the starter workflows, we do not know what is the token permission set at the org/ repo level for whoever will use the workflow. So, the permissions should be specified in the workflow file.

Reply:
> So, if contents: read is not added in the workflow file, the token will get all permissions.

That's a good point in case of permissive repos/orgs.

Reply:
However, given that we have already specified a workflow level read permission, this job level permission seems redundant.
I am not against good practice of explicit permissions, but this one here is just adding one more line which is not much useful. Simplicity of workflow is important, especially in this repo as it is meant for the users to get started with workflows.

Reply:
@bishal-pdMSFT this is good feedback. Since the changes are done via automation, I would want to address this via a configuration option in the automation. What if there is a configuration option to `Only set job level permissions`. In this case, the workflow level (top level) permissions will not be set, and only job level permissions will be set. So the workflow would be simpler. Moreover, the workflow would get a score of 9 in scorecards. Not perfect from a security perspective, but it balances security with simplicity. Will this work?

CC: @laurentsimon

Reply:
Can we add smartness in automation to `Set workflow level permissions` and `Set job level permissions if they are not same as default`?

Reply:
@bishal-pdMSFT did you mean `Set job level permissions if they are not same as workflow level permissions`? As of now, the automation sets the workflow level permissions to `read-all`. So, in this case, if the job level permission is not set, it would be `read-all`, which is more than `contents: read` (current value). `read-all` will include all read permissions, including `SecurityEvents` - example [here](https://github.com/varunsh-coder/actions-playground/runs/4844452755?check_suite_focus=true#step:1:13).

Reply:
I guess that should be fine atleast for a GitHub support action like `actions/checkout`. A job containing a third party action may decide to restrict using `contents: read`.
Again it is a trade off between security and simplicity.

Reply:
@bishal-pdMSFT to clarify, are you suggesting only having the `permissions: read-all` permission at workflow level, and removing the permissions from job level? If that is the case, the configuration could be something like `Only add job level permissions if they are higher than read-all`.

Reply:
if there's no write permission needed for the job, the `read-all` top-level permission does simplify the workflow, I agree. You'd still get scorecard 10, correct?

Reply:
@laurentsimon please see feedback on [this PR](https://github.com/Automattic/vip-go-mu-plugins/pull/2815#discussion_r786933173). The ask is to set `contents: read` at the workflow (top) level instead of `permissions: read-all`. This is what the repo level restrictive setting is (screenshot [here](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#configuring-the-default-github_token-permissions)), so it would align with that setting. Then I can update the StepSecurity automation such that if job level permission is `contents: read`, then do not add it explicitly (inherit from workflow). I believe this is aligned with what @bishal-pdMSFT is also asking for. What are your thoughts? I don't think you need to make changes to scorecards for this.

Reply:
`contents: read` sgtm.  If additionally read permissions are needed for a job within the workflow, you'll add them all at the job level, correct?

Reply:
> `contents: read` sgtm. If additionally read permissions are needed for a job within the workflow, you'll add them all at the job level, correct?

Yes. Thanks! I will make this change.

Reply:
@bishal-pdMSFT the logic has been updated to add `contents: read` at workflow level, and if job needs `contents: read` only, then to set no permissions at the job level. Applied changes to this yaml file. Let me know if the update looks fine.

Reply:
@bishal-pdMSFT - reminder to please review this. Thanks!